spring-boot-project/spring-boot-docs/src/docs/asciidoc/messaging/pulsar.adoc

```diff
@@ -0,0 +1,144 @@
+
+TIP: The Reactive support is built on top of the imperative support and therefore the reactive starter includes the imperative starter.
+As such, the imperative components are auto-configured and available in both reactive and imperative scenarios.
+
+[[messaging.pulsar.connecting]]
+=== Connecting to Pulsar
+When you use the Pulsar starter, Spring Boot will auto-configure and register a `PulsarClient` bean.
+
+By default, the application tries to connect to a local Pulsar instance at `pulsar://localhost:6650`.
+This can be adjusted by setting the `spring.pulsar.client.service-url` property to a different value.
+
+TIP: The value must be a valid https://pulsar.apache.org/docs/client-libraries-java/#connection-urls[Pulsar Protocol] URL
+
+[[messaging.pulsar.connecting.auth]]
+==== Authentication
+To connect to a Pulsar cluster that requires authentication, you need to specify which authentication plugin to use by setting the `authPluginClassName` and any parameters required by the plugin.
+You can set the parameters as a single JSON-encoded string or as map of parameter names to parameter values.
+The following listings show both approaches:
+
+[source,yaml,indent=0,subs="verbatim",role="primary"]
+.[.small]#Map#
+----
+spring:
+  pulsar:
+    client:
+      auth-plugin-class-name: org.apache.pulsar.client.impl.auth.oauth2.AuthenticationOAuth2
+      authentication:
+        issuer-url: https://auth.server.cloud/
+        private-key: file:///Users/some-key.json
+        audience: urn:sn:acme:dev:my-instance
+----
+
+.[.small]#JSON encoded string#
+[source,yaml,indent=0,subs="verbatim",role="secondary"]
+----
+spring:
+  pulsar:
+    client:
+      auth-plugin-class-name: org.apache.pulsar.client.impl.auth.oauth2.AuthenticationOAuth2
+      auth-params: "{\"privateKey\":\"file:///Users/some-key.json\",\"issuerUrl\":\"https://auth.server.cloud/", \"audience\":\"urn:sn:acme:dev:my-instance"}"
+----
+
+TIP: Using a map is the recommended approach as it is less error-prone and easier to read.
+
+For complete details on the client and authentication see the Spring Pulsar {spring-pulsar-docs}#pulsar-client[reference documentation].
+
+[[messaging.pulsar.connecting-reactive]]
+=== Connecting to Pulsar (Reactively)
+When the Reactive auto-configuration is activated, Spring Boot will auto-configure and register a `ReactivePulsarClient` bean.
+
+The `ReactivePulsarClient` adapts an instance of the previously described `PulsarClient`.
+Therefore, use the properties described previously (`spring.pulsar.client.*`) to configure the `ReactivePulsarClient`.
+
+[[messaging.pulsar.admin]]
+=== Connecting to Pulsar Administration
+Spring Pulsar's `PulsarAdministration` client is also auto-configured.
+
+By default, the application tries to connect to a local Pulsar instance at `\http://localhost:8080`.
+This can be adjusted by setting the `spring.pulsar.administration.service-url` property to a different value in the form `(http|https)://<host>:<port>`.
```

Comment:
Please use `configprop:spring.pulsar.administration.service-url[]` as this ensures that the property exists.